app/Controllers/userController.php

```diff
@@ -243,10 +247,12 @@ public static function createUser(string $new_user_name, ?string $email, string
 			$ok &= !file_exists($configPath);
 		}
 		if ($ok) {
-			if (!is_dir($homeDir)) {
-				mkdir($homeDir, 0770, true);
+			// $homeDir must not exist beforehand,
+			// otherwise it might be multiple remote parties racing to register one username
+			$ok = mkdir($homeDir, 0770, true);
```

Comment:
```suggestion
			$ok = @mkdir($homeDir, 0770, true);
```

Would otherwise generate a warning if the directory already exists

Reply:
It makes sense to silence these warnings, but I don't think we want to silence them at the call level here.
- It is a good practice to set `display_errors = Off` in php.ini in production
- The official docker image has that setting
- Minz FrontController sets that setting so, unless `FRESHRSS_ENV` is set to `development`.
- Silencing individual standard function calls would break the way `FRESHRSS_ENV=development` is expected to work
- Silencing individual standard function calls adds cognitive load to comprehend the code

Reply:
The thing is that we catch the error already, and react to it, so there is not much more to do about it. With the same reasoning, you would have to re-throw every error again in a `try/catch` construct.
I do run with `FRESHRSS_ENV=development` and my expectation is to be visually alerted (and break) in the case of error conditions not accounted for, which is not the case here.
But the situation is likely so rare that I do not really mind

Reply:
I see what you mean - the situation is handled in code, so it's not something the admin has to be notified and do anything about.